src/GraphQL/Types/Schema.cs

```diff
@@ -34,13 +35,58 @@ public Schema()
         /// <summary>
         /// Create an instance of <see cref="Schema"/> with a specified <see cref="IServiceProvider"/>, used
         /// to create required objects.
+        /// Pulls registered <see cref="IConfigureSchema"/> instances from <paramref name="services"/> and
+        /// executes them.
         /// </summary>
         public Schema(IServiceProvider services)
+            : this(services, true)
+        {
+        }
+
+        /// <summary>
+        /// Create an instance of <see cref="Schema"/> with a specified <see cref="IServiceProvider"/>, used
+        /// to create required objects.
+        /// If <paramref name="runConfigurations"/> is <see langword="true"/>, pulls registered
+        /// <see cref="IConfigureSchema"/> instances from <paramref name="services"/> and executes them.
+        /// </summary>
+        public Schema(IServiceProvider services, bool runConfigurations = true)
```

Comment:
Why do you think this ctor will need? Why not just add one below with `IEnumerable<IConfigureSchema> configurations`?

Reply:
Also I doubt that `public Schema(IServiceProvider services)` ctor needs to run configurations. This somewhat contradicts the design of dependency injection. I suspect that you did it so as not to force the users to add new or modify existing ctors adding `IEnumerable<IConfigureSchema> configurations` parameter. 

This is a question of design. Well... look at `Field Middleware` documentation page:
![изображение](https://user-images.githubusercontent.com/21261007/147154227-e6c518b1-6ecd-4033-8455-881ada80f8d5.png)

We **could** pull middlewares from the serviceprovider. But we did not in favor of explicit dependency injection (which is good decision in my opinion). I believe that the presence of `IServiceProvider` in ctor in some sense _prevents_ good design. The temptation to pull something out of it constantly arises. But this was the initial design, it is now difficult to change it (if at all necessary).

So my advise here to make dependencies explicit and not pull them out inside from serviceprovider.

Reply:
Thus, only two constructors will remain and the first will call last with null of empty array.

Reply:
I was trying to allow someone to derive from Schema without running any configurations set up by the builder.  Also I was attempting to maintain ease of use and backwards compatibility - that's why it pulls the configurations from the service provider in the typical constructor `public Schema(IServiceProvider services)`.  I'm open to reviewing changes here, with those points in mind.

Note: for some of these changes that require review, let's make separate PRs.

Reply:
I am skeptical about having `public Schema(IServiceProvider services)` NOT run any configurations.  I feel that it would be very easy for implementations to write `public MySchema(IServiceProvider services) : base(services)` and forget that in v5 they need the IEnumerable.  Or they don't need it at the time.  Then they start raising issues here asking why certain extension methods do not work properly.  And we have to repeatedly explain that they are using old setup code that does not work in v5 with the current documentation.  (And which worked fine in v4.6, but with no change to their code, it mysteriously stops working.)

Reply:
I'll make changes for my review here. OK? `develop` branch?

Reply:
I completely understand that this is abnormal use of IServiceProvider however.  Perhaps we remove that particular constructor entirely and force people to use another.  But changing the behavior in v5 like that isn't going to be an immediately noticeable problem, like renaming a method would.  It will be hard to track down -- it will probably just be a mystery why certain extension methods don't appear to be working anymore.

Reply:
> I'll make changes for my review here. OK? `develop` branch?

Yes, but please create separate PRs for topics like this that need further discussion, so we can merge changes that do not need discussion.

Reply:
`IServiceProvider` in `Schema` ctor is a design flaw in my opinion. Everything else is only the consequences of this. I would not want to exacerbate the situation and continue to pull out dependencies from it.

Reply:
> Well... look at `Field Middleware` documentation page

Entirely the kind of code that the builder system seeks to avoid.

Reply:
Buidler system may well avoid it (I have not watched yet). Another question is, whether it is necessary to avoid it. The question is what is considered right and what is not.

Reply:
> `IServiceProvider` in `Schema` ctor is a design flaw in my opinion. Everything else is only the consequences of this. I would not want to exacerbate the situation and continue to pull out dependencies from it.

If there is a way to eliminate the 'flaw' we can explore it.  I think the system works well, even if it is not the typical way DI is set up.  Also, I do not think it is necessarily abnormal.  How do you think ASP.NET Core works?  It takes the service provider and uses it to create installed middleware and all sorts of objects.  Controllers, for example, if registered in DI will pull from DI, and if not will be created by the DI engine.  It is not a 'design flaw' in ASP.Net Core that every single piece of middleware and controller is directly referenced in a derived type.  It is a design benefit.

Reply:
Perhaps to "fix" the "flaw" we can explore having the Schema accept another interface which is a wrapper around the service provider, or something.  For example, we could define `ISchemaTypes`.  The `AddGraphQL` method can set up this interface (in a sense, internal to GraphQL.NET) and Schema then would require `ISchemaTypes` instead of `IServiceProvider`.  This of course would need to be explored to see how much sense it makes.  Of course `SchemaTypes` still needs `IServiceProvider` to create its required instances.

Reply:
> Builder system may well avoid it (I have not watched yet). Another question is, whether it is necessary to avoid it. The question is what is considered right and what is not.

You should not start creating PRs and criticizing the design until you understand the full effect and usefulness of this builder system.  It works VERY well.  Sure maybe we want to rename some interfaces or move methods to other namespaces, but we should not be removing functionality.

Reply:
I know and understand. I just don't want "to be ASP.NET Core".

Reply:
#2776